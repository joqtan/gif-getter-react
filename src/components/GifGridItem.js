import React from 'react'
import 'animate.css'
import PropTypes from 'prop-types'

const GifGridItem = ({ title, url }) => {
    return (
        <div className='card animate__animated animate__fadeInUpBig '>
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}

export default GifGridItem
