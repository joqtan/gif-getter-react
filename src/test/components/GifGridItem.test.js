import { shallow } from 'enzyme'
import GifGridItem from '../../components/GifGridItem'

describe('tests on <GifGridItem/> component', () => {
    const title = 'title'
    const url = 'http:localhost/image.jpg'
    const wrapper = shallow(<GifGridItem title={title} url={url} />)
    test('should show the component properly', () => {
        expect(wrapper).toMatchSnapshot()
    })
    test('p should have a paragraph with the title', () => {
        const p = wrapper.find('p')
        expect(p.text().trim()).toBe(title)
    })
    test('img should have a image with source and alt from props', () => {
        const img = wrapper.find('img')
        expect(img.prop('src')).toBe(url)
        expect(img.prop('alt')).toBe(title)
    })
    test('div should contain a propertie called animate__fadeInUpBig', () => {
        const div = wrapper.find('div')
        expect(div.hasClass('animate__fadeInUpBig')).toBe(true)
    })
})
