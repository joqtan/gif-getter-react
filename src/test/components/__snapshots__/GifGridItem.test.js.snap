// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <GifGridItem/> component should show the component properly 1`] = `
<div
  className="card animate__animated animate__fadeInUpBig "
>
  <img
    alt="title"
    src="http:localhost/image.jpg"
  />
  <p>
    title
  </p>
</div>
`;
